docs/self-hosting.md

```diff
@@ -0,0 +1,43 @@
+Schism is a self hosted compiler. Each revision of Schism is built by a
+previous revision of Schism. It's sometimes important to understand how this
+works in order to add new functionality to Schism, as it affects when you are
+allowed to use certain features. Schism has four stages: stage0, stage1,
+stage2, and stage3.
+
+Stage0 is also called the snapshot. We keep it checked into the repo, but it
+doesn't have to be (and probably shouldn't be). It's a working version of the
```

Comment:
I think it makes sense to check in the binary.  It allows Schism to prove that it's self-hosting.  It also allows comparison between the bootstrapped binary and an externally-compiled binary.  It's convenient too when you don't have an external Scheme handy.

So, consider changing the sentence to: "We keep it checked into the repo."

Reply:
Sounds good, I made the change.

My reason for saying we shouldn't check it in is more because Git is not great for tracking binaries. I was thinking ideally we'd have the snapshot stored in some other well known location (an S3 bucket, for example), that we periodically push new snapshots to. Rust, for example, does this. For now though, the snapshot is small and it's not worth building a ton of extra infrastructure to do it.

One nice feature about keeping it in the repo is that it is really easy to tell that the snapshot matches what it should be (i.e. the previous commit compiled by the snapshot in the previous commit).